android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -528,4 +583,102 @@ private String signalingStateString(PeerConnection.SignalingState signalingState
         }
         return null;
     }
+
+    /**
+     * Serialization and Parsing helpers
+     */
+
+    private WritableMap serializeStream(String streamReactTag, MediaStream stream) {
+
+        Log.d(TAG, "streamReactTag = " + streamReactTag);
+        WritableMap params = Arguments.createMap();
+        params.putInt("id", id);
+        params.putString("streamId", stream.getId());
+        params.putString("streamReactTag", streamReactTag);
+
+        WritableArray tracks = Arguments.createArray();
+
+        for (VideoTrack track: stream.videoTracks) {
+            tracks.pushMap(serializeTrack(track));
+        }
+        for (AudioTrack track: stream.audioTracks) {
+            tracks.pushMap(serializeTrack(track));
+        }
+
+        params.putArray("tracks", tracks);
+
+        return params;
+    }
+
+    public String serializeDirection(RtpTransceiver.RtpTransceiverDirection src) {
+        switch(src) {
+            case INACTIVE:
+                return "inactive";
+            case RECV_ONLY:
+                return "recvonly";
+            case SEND_ONLY:
+                return "sendonly";
+            case SEND_RECV:
+                return "sendrecv";
+            default:
+                throw new Error("Invalid direction");
+        }
+    }
+
+    private ReadableMap serializeTrack(MediaStreamTrack track) {
+        WritableMap trackInfo = Arguments.createMap();
+        trackInfo.putString("id", track.id());
+        trackInfo.putInt("peerConnectionId", this.id);
+        if (track.kind().equals("video")) {
```

Comment:
I'd say we leave this out of native and just set it to the empty string in JS

Reply:
Done